docs/book/src/common-collections/storage_map.md

```diff
@@ -80,3 +82,39 @@ storage {
 
 storage.w.map1.insert(..);
 ```
+
+### Storage maps within storage maps
+
+It is currently not possible to use somthing like StorageMap<K, StorageVec> directly in Sway. The workaround for this is to use a Sway reference library called [StorageMapVec](https://github.com/FuelLabs/sway-libs/tree/master/libs/storagemapvec) such as below:
+
+```sway
+use storagemapvec::StorageMapVec;
+
+storage {
+    mapvec: StorageMapVec<u64, u64> = StorageMapVec {},
+}
+
+abi MyContract {
+    #[storage(read, write)]
+    fn push(key: u64, value: u64);
+
+    #[storage(read)]
+    fn get(key: u64, index: u64);
+}
+
+impl MyContract for Contract {
+    #[storage(read, write)]
+    fn push(key: u64, value: u64) {
+        // this will push the value to the vec, which is accessible with the key
+        storage.mapvec.push(key, value);
+    }
+
+    #[storage(read)]
+    fn get(key: u64, index: u64) {
+        // this will retrieve the vec at given key, and then retrieve the value at given index from that vec
+        storage.mapvec.get(key, index)
+    }
+}
```

Comment:
Would it be possible to put this example under the examples directory and `#include` it as we do in other places?

Reply:
can do! will make a new commit for it :) thanks!

Reply:
Closed this and created a new PR [here](https://github.com/FuelLabs/sway/pull/3857).